spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -454,18 +458,68 @@ def change_mpl_backend(self, command):
                 self.silent_execute(command)
 
     # ---- Spyder-kernels methods -------------------------------------------
+    def get_editor(self, filename=None):
+        """Get editor for filename and set it as the current editor."""
+        editorstack = self.get_editorstack()
+        if editorstack is None:
+            return None
+
+        if not filename:
+            return editorstack.get_current_editor()
```

Comment:
Why this? I don't like to use the current editor as the default editor if not `filename` is passed. Instead, I think `filename` should be mandatory.

Reply:
Well if I call cell_count(), what I probably want to do is to count the number of cells in the current file.

Reply:
So in general, this is probably what should happen.

Reply:
It is nice to not have to type whole filename in the console to run a cell. If I run `runcell('cellname')` in the console I would expect it to run the cell called 'cellname' from the current file.

However, I think this functionally might be better placed in `handle_run_cell()`. 

It would also be handy if `runcell()` with no aguments would run the current cell(where the cursor is/was. The cell that is highlighted) from the current editor.

Reply:
> It is nice to not have to type whole filename in the console to run a cell

Ok, fair enough.

Reply:
The kernel will now request the filename from the frontend before calling this function.

Reply:
Ok, thanks. That makes more sense to me.